website/content/docs/v0.15/Guides/upgrading-talos.md

```diff
@@ -57,3 +62,71 @@ future.
+(Talos ensures the etcd cluster is healthy and will remain healthy after our node leaves the etcd cluster, before allowing a control plane node to be upgraded.)
+
+Once all the processes are stopped and the services are shut down, the filesystems will be unmounted.
+This allows Talos to produce a very clean upgrade, as close as possible to a pristine system.
+We verify the disk and then perform the actual image upgrade. We set the bootloader to boot _once_ with the new kernel and OS image, then we reboot.
+
+After the node comes back up and Talos verifies itself, it will make 
+the bootloader change permanent, rejoin the cluster, and finally uncordon itself to receive new workloads.
+
+### FAQs
+
+**Q.** What happens if an upgrade fails?
+
+**A.** Talos Linux attempts to safely handle upgrade failures.
+
+The most common failure is an invalid installer image reference.
+In this case, Talos will fail to download the upgraded image and will abort the upgrade.
+
+Sometimes, Talos is unable to successfully kill off all of the disk access points, in which case it cannot safely unmount all filesystems to effect the upgrade.
+In this case, it will abort the upgrade and reboot. (`upgrade --stage` can ensure that upgrades can occur even when the filesytems cannot be unmounted.)
+
+It is possible (especially with test builds) that the upgraded Talos system will fail to start.
+In this case, the node will be rebooted, and the bootloader will automatically use the previous Talos kernel and image, thus effectively rolling back the upgrade.
+
+Lastly, it is possible that Talos itself will upgrade successfully, start up, and rejoin the cluster but your workload will fail to run on it, for whatever reason.
+This is when you would use the `talosctl rollback` command to revert back to the previous Talos version.
+
+**Q.** Can upgrades be scheduled?
+
+**A.** Because the upgrade sequence is API-driven, you can easily tie it in to your own business logic to schedule and coordinate your upgrades.
+
+**Q.** Can the upgrade process be observed?
+
+**A.** Yes, using the `talosctl logs --follow machined` command.
+
+**Q.** Are worker node upgrades handled differently from control plane node upgrades?
+
+**A.** Short answer: no.
+
+Long answer:  Both node types follow the same set procedure.
+From the user's standpoint, however, the processes are identical. However, since control plane nodes run additional services, such as etcd, there are some extra steps and checks performed on them. 
+For instance, Talos will refuse to upgrade a control plane node if that upgrade would cause a loss of quorum for etcd.
+If running a single-node cluster, and you want to force an upgrade despite the loss of quorum, you can set `preserve` to `true`. 
+
+**Q.** Will an upgrade try to do the whole cluster at once?
+Can I break my cluster by upgrading everything?
+
+**A.** No.
+
+Nothing prevents the user from sending near-simultaneous upgrades to each node of the cluster.
```

Comment:
I think this needs further clarification, as upgrading every node at the same time might break stateful applications like Rook/Ceph. I wouldn't recommend during simultaneous upgrades in the docs. Even though it's possible to do such upgrades, I feel we shouldn't recommend that.

Reply:
Clarificaiton added. (I just copy/pasted this from the other doc, so glad you are reviewing this!)